src/openrct2/entity/Yaw.hpp

```diff
@@ -42,12 +56,26 @@ namespace OpenRCT2::Entity::Yaw
     {
         return yaw;
     }
-    [[nodiscard]] constexpr int32_t YawToPrecision(int32_t yaw, uint8_t precision)
+
+    [[nodiscard]] constexpr int32_t YawToPrecision(int32_t yaw, SpritePrecision precision)
     {
-        return yaw >> PrecisionOffset[precision];
+        return yaw >> PrecisionOffset[static_cast<uint8_t>(precision)];
     }
-    [[nodiscard]] constexpr uint8_t NumSpritesPrecision(uint8_t precision)
+
+    [[nodiscard]] constexpr uint8_t NumSpritesPrecision(SpritePrecision precision)
+    {
+        return (1 << static_cast<uint8_t>(precision)) >> 1;
+    }
+
+    // Converts a power of 2 to a sprite precision value
+    [[nodiscard]] constexpr SpritePrecision PrecisionFromNumSprites(uint8_t numSprites)
```

Comment:
You could use bitscanforward which would be more efficient we have helper functions for it. I would drop the existence of this function in the header as its only used in one place so just have it defined there.

```
// It is assumed that numSprites is a non zero power of 2
static constexpr SpritePrecision PrecisionFromNumSprites(uint8_t numSprites)
{
    return static_cast<SpritePrecision>(bitscanforward(numSprites) + 1);
}
```